Square/src/Square/Entities/Sprite.cpp

```diff
@@ -109,37 +130,61 @@ namespace Square {
 
 	void Sprite::Render()
 	{
-		Vector2 pos = Pos(world);
-		Vector2 scale = Scale(world);
+		if (mTexture)
+		{
+			Vector2 pos = Pos(world);
+			Vector2 scale = Scale(world);
+
+			mRenderRect.x = (int)(pos.x - mWidth * scale.x * 0.5f);
+			mRenderRect.y = (int)(pos.y - mHeight * scale.y * 0.5f);
+
+			mRenderRect.w = (int)(mWidth * scale.x);
+			mRenderRect.h = (int)(mHeight * scale.y);
+
+			if (mFlash && mFlashFrames >= 0)
+			{
+				SDL_SetTextureColorMod(mTexture.get(), mFlashColor.r, mFlashColor.g, mFlashColor.b);
+				SDL_SetTextureAlphaMod(mTexture.get(), mFlashColor.a);
+				mFlashFrames++;
+			}
+			else
+			{
+				SDL_SetTextureColorMod(mTexture.get(), mColor.r, mColor.g, mColor.b);
+				SDL_SetTextureAlphaMod(mTexture.get(), mColor.a);
+				if (mFlashFrames < 0) mFlashFrames++;
+			}
+
+			mGraphics.DrawTexture(mTexture, (mClipped) ? &mClipRect : nullptr, &mRenderRect, Rotation(world), mFlip);
+
+			if (mFlash && mFlashFrames >= FLASH_FRAMES)
+			{
+				mFlash = false;
+				mFlashFrames = -FLASH_FRAMES * 2.5f;
+			}
+
+			SDL_SetTextureColorMod(mTexture.get(), 255, 255, 255);
+		}
+	}
+
+	Sprite& Sprite::operator=(Sprite other)
```

Comment:
thxs I definitely wouldn't have figured this out lol

Reply:
Google "assignment operator" or "copy constructor" if you're curious about it